src/chfn.c

```diff
@@ -230,6 +232,10 @@ static char *copy_field (char *in, char *out, char *extra)
 			break;
 
 		if (NULL != extra) {
+			if (strlen(extra) + 1 + strlen(f) + 1 > out_size) {
+				return NULL;
+			}
+
 			if (!streq(extra, "")) {
 				strcat (extra, ",");
 			}
```

Comment:
Why is the check here, if the out_size is the size protecting the strcpy() in line 248?